### PR TITLE
[@types/leaflet-routing-machine] fixing return types in createMarker function

### PR DESCRIPTION
**Pull request by augustoostapechen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [L.Routing.Plan](https://www.liedman.net/leaflet-routing-machine/api/#l-routing-plan)

### CODE REVIEW

File: types/leaflet-routing-machine/leaflet-routing-machine-tests.ts [hunk 1]
```diff
@@ -7,9 +7,24 @@ L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
...
+const waypoints = [
+    L.latLng(57.75934, 11.93927),
+    L.latLng(57.74, 11.94),
+    L.latLng(57.6792, 11.949),
+];
+
+const myPlan = new L.Routing.Plan(waypoints, {
+    addWaypoints: false,
+    createMarker: (index: number, waypoint: L.Routing.Waypoint, numberWaypoints: number) => {
+      if (index === 0 || index === numberWaypoints - 1) {
```
Comment (chanakadrathnayaka) [inline types/leaflet-routing-machine/leaflet-routing-machine-tests.ts:19]: Thanks for updating the definition. Do you think this is a good example of using `createMarker` function? What if we update this function as,
```
if(waypoint && index > -1 && index < numberOfWaypoints){
    return L.marker(waypoint.latLng);
}
return false;
```
Please share your opinion.

File: types/leaflet-routing-machine/index.d.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ declare module 'leaflet' {
             waypointNameFallback?: ((latLng: LatLng) => string) | undefined;
             createGeocoder?: ((waypointIndex: number, numberWaypoints: number, plan: Plan) => {}) | undefined;
             addButtonClassName?: string | undefined;
-            createMarker?: ((waypointIndex: number, waypoint: Waypoint, numberWaypoints: number) => Marker) | undefined;
+            createMarker?: ((waypointIndex: number, waypoint: Waypoint, numberWaypoints: number) => Marker | boolean) | undefined;
```
Comment (chanakadrathnayaka) [inline types/leaflet-routing-machine/index.d.ts:92]: Let's update `numberWaypoints` as `numberOfWaypoints`.

**Documentation**
Creates a marker to use for a waypoint. The function should have the signature createMarker(<Number> i, <[L.Routing.Waypoint](https://www.liedman.net/leaflet-routing-machine/api/#l-routing-waypoint)> waypoint, <Number> n), where i is the waypoint’s index, waypoint is the waypoint itself, and n is the total _number of waypoints_ in the plan; if return value is falsy, no marker is added for the waypoint

**Review by chanakadrathnayaka — CHANGES_REQUESTED** [review]

File: types/leaflet-routing-machine/leaflet-routing-machine-tests.ts (hunk 1 above)
Comment (augustoost) [reply types/leaflet-routing-machine/leaflet-routing-machine-tests.ts:19]: That's works as well! I will update the function.

File: types/leaflet-routing-machine/index.d.ts (hunk 2 above)
Comment (augustoost) [reply types/leaflet-routing-machine/index.d.ts:92]: Okay, in the createGeocoder function and others too?

File: types/leaflet-routing-machine/leaflet-routing-machine-tests.ts (hunk 1 above)
Comment (chanakadrathnayaka) [reply types/leaflet-routing-machine/leaflet-routing-machine-tests.ts:19]: Awesome. Thanks

File: types/leaflet-routing-machine/index.d.ts (hunk 2 above)
Comment (chanakadrathnayaka) [reply types/leaflet-routing-machine/index.d.ts:92]: If you have time :) Thanks a lot

File: types/leaflet-routing-machine/index.d.ts [hunk 3]
```diff
@@ -192,7 +192,7 @@ declare module 'leaflet' {
         }
 
         class GeocoderElement {
-            constructor(wp: Waypoint, i: number, numberWaypoints: number, options: GeocoderElementOptions);
+            constructor(waypoint: Waypoint, i: number, numberOfWaypoints: number, options: GeocoderElementOptions);
```
Comment (augustoost) [inline types/leaflet-routing-machine/index.d.ts:195]: numberOfWaypoints

Comment (augustoost) [inline types/leaflet-routing-machine/index.d.ts:195]: I change the wp variable too :)

**Review by chanakadrathnayaka — APPROVED** [review]
LGTM

**Pull request merged**